src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -517,4 +519,23 @@ public int hashCode() {
     public String toString() {
         return "transaction " + hash.toString();
     }
+
+    public long addBatchTxnCount(Tangle tangle) {
+        byte[] tritsSig = getSignature();
+        String trytesSig = Converter.trytes(tritsSig);
+        String asciiSig = Converter.trytesToAscii(trytesSig);
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode rootNode = objectMapper.readTree(asciiSig);
+            JsonNode idNode = rootNode.path("tx_num");
+            long txnCount = idNode.asLong();
+
+            tangle.addBatchCount(txnCount - 1);
```

Comment:
I think this (txnCount - 1) logic includes some hidden assumption, which is not so clear to the readers of this piece of code.